File: python/1_two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        M = {}
        for i, n in enumerate(nums):
            M[n] = i
        for i, n in enumerate(nums):
            if M.get(target - n) and M[target - n] != i:
                return [i, M[target - n]]

File: cpp/1_two-sum.cpp
class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        unordered_map<int,int> m;
        vector<int> result;
        for(int i=0;i<nums.size();i++)
        {
            m[nums[i]]=i;            
        }
        for(int i=0;i<nums.size();i++)
        {
            int othernum=target-nums[i];
            if(m[othernum]&&m[othernum]!=i)
            {
                result.push_back(i);
                result.push_back(m[othernum]);
                break;
            }                
        }
        return result;
    }
};